### PR TITLE
Updating README with user-relevant GridItem props; passing itemsLength to GridItem

### PR DESCRIPTION
**Pull request by rovolution**
hi @jrowny,

Wanted to start off by saying thank you for this awesome component!

Submitting this PR for the following:
- Updating the documentation to display relevant GridItem props that I think the user should be aware of
- Passing `itemsLength` down as a prop to GridItem. I ran into a scenario where I needed to apply the z-index in reverse order on the GridItems, and I needed the total length of the data set to derive the z-indicies

### CODE REVIEW

File: lib/AbsoluteGrid.jsx [hunk 1]
```diff
@@ -60,6 +60,7 @@ export default class AbsoluteGrid extends React.Component {
       var gridItem = React.cloneElement(this.props.displayObject, _.assign(this.props.displayObject.props, {
         style: style,
         item: item,
+        itemsLength: this.props.items.length,
```
Comment (jrowny) [inline lib/AbsoluteGrid.jsx:63]: If we `var len = this.props.items.length;` before the map statement on about line 54, then there's a very minor efficiency enhancement.

**Comment by jrowny** [comment]
Thanks so much @rovolution! I did make one comment but it's very minor so I won't let it hold up the PR if you're not able to get to it, I'll merge it in later today.

File: lib/AbsoluteGrid.jsx (hunk 1 above)
Comment (rovolution) [inline lib/AbsoluteGrid.jsx:63]: i believe JS arrays maintain the `length` value as values are added/removed and do not compute it on each access, but I can make that change.

Comment (jrowny) [inline lib/AbsoluteGrid.jsx:63]: You're right, I just ran a jsperf and it's identical. Going to go ahead and merge!

Comment (rovolution) [inline lib/AbsoluteGrid.jsx:63]: thanks!

**Pull request merged**